src/Microsoft.AspNet.Mvc.ModelBinding/Binders/CompositeModelBinder.cs

```diff
@@ -97,7 +97,17 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
                 var result = await binder.BindModelAsync(bindingContext);
                 if (result != null)
                 {
-                    return result;
+                    // Use returned ModelBindingResult if it either indicates model was set or is related to errors.
+                    // The second condition is necessary because some binders add errors that would be duplicated when
```

Comment:
which "this"?  the case where `!IsModelSet && bindingContext.ModelState.ContainsKey(bindingContext.ModelName) && thatModelState.IsValid`?  all I know is it's freaking hard to remove the `bindingContext.ModelState.ContainsKey(bindingContext.ModelName)` bit and get the 10s of failing tests working again.  not sure if any are hitting that particular case.

and is that really part of this PR?  I didn't _add_ cases like that or even change anything in `TypeConverterModelBinder` or `TypeMatchModelBinder` except to name parameters.  we're talking about whether or not to return `null` instead of a slightly odd result that would have propagated up before I started on these bugs.